src/config.rs

```diff
@@ -180,16 +180,17 @@ impl Default for DbOptions {
     }
 }
 
+#[allow(unreachable_code)]
 fn default_block_cache() -> Option<Arc<dyn DbCache>> {
-    if cfg!(all(feature = "moka", feature = "foyer")) {
-        Some(Arc::new(crate::db_cache::moka::MokaCache::new()))
-    } else if cfg!(feature = "foyer") {
-        Some(Arc::new(crate::db_cache::foyer::FoyerCache::new()))
-    } else if cfg!(feature = "moka") {
-        Some(Arc::new(crate::db_cache::moka::MokaCache::new()))
-    } else {
-        None
+    #[cfg(feature = "moka")]
```

Comment:
this will fail to compile if both features are enabled right?

Reply:
No, it still compiles. Both blocks are included, but since moka is the preferred default, it's returned and the foyer code is treated as dead (hence the allow dead code)

Reply:
Or are you suggesting I use a `#[cfg(all(feature = "moka", feature = "foyer"))]` because it will return `None` if both foyer and moka are set?

Reply:
No, I thought you'd get a compile error due to the dead code. Nvm!